### DISCUSSION TITLE
Why things don't add up in the app - the real reason (and fix)

### DISCUSSION THREAD

**Discussion by gekkehenkie11** [emporia-vue-local/esphome#287] [General]
Over the years there's been a lot of discussion on the Emporia forum about why things don't add up in the app. It first was theorized by some users that the Emporia VUE probably would measure apparent power instead of real power. Which seemed to make sense: according to my VUE2, my Air conditioner would use about 800 Watt running but 120 Watt standy. I tested with some other devices that all came to about 6-8 Watt standby and a power factor near 0 in standby mode. So I thought the Emporia indeed measured apparent power for a long time.

Then I started looking into making an open source firmware for the Emporia Atmel chip: https://github.com/emporia-vue-local/esphome/discussions/173 and to my surprise I learned that the Emporia actually was measuring real power. In the world of samples, real power is simply V * I and then averaging that, what they indeed did! Then I called in  the help of my brother who has a major in electrical engineering. I dumped buffers of voltage and current samples and supplied them to him. He first theorized, based on these samples, that it was a timing problem within the Emporia. But then when I supplied more samples, it seemed there were some problems that didn't make sense and couldn't be explained that way. Then he asked me to describe EXACTLY what I've been measuring and that finally led to the answer :) So here we go:

The problem arises for 3 phase users, however ALSO for some specific 2 phase users in North America. Our own hero @flaviut actually is one of those, going by this post: 

https://github.com/emporia-vue-local/emporia-vue2-reversing/issues/1#issuecomment-1088202075

Because this is the exact problem! Whenever somebody's electrical system has 120 degrees shift between the voltages and if they then supply 2 phases to a breaker, for example to connect a 220-240V air conditioner things go wrong...

Here on Aruba we have 3 phases: 127V between neutral and phase and 220V (and 120 degrees) between the phases. So the thing is, that THIS is what happens to the resulting voltage when you connect a breaker to two phases:

![WhatsApp Image 2024-06-01 at 6 07 59 AM](https://github.com/emporia-vue-local/esphome/assets/10788019/124cc5a5-8d39-4980-bef2-f1d6d99a9260)

The resulting Voltage has a phase shift of 30 degrees and a multiplier of V3 (=1.73). So in this case you can not simply do V * I per sample and then simply use a 2.0 multiplier!! The correct way to calculate it in this scenario is:

(V1 - V2) * I

So we should take the difference between the two voltages and multiply THAT with the current!! 

With this modification my Emporia VUE2 all of a sudden becomes super accurate :) I get close to 6W now in standby and close to 710 Watt when the air conditioner is on which is really close to what it indeed is (before, with multiplier 2.0 I got close to 800-830 which is too high).

*EDIT* There's actually a super simple alternative to this. Just use 2 breakers and add the power values up! With Emporia stock firmware this would never work since it would select the wrong phases, but I just tried it and if you lock the right phases this indeed works too! So you would just need an extra CT if you use 2 phases per breaker. Make sure that you don't use filters like 'abs' or 'pos' though, since one leg actually might become negative by design! For example my air conditioner in standby yields +30W on one leg and -22W on the other, so 8W in standby which is correct. 

That's it!! I can't thank enough my brother for dedicating his time to help figure this out. Without him I'm sure I still would be banging my head against the wall about all this, haha :) Thanks bro :)

**Comment by g3gg0** [comment]
Great progress.
I wonder if the protocol you implemented is the same as the original one.
Didnt Emporia send the power calculation result against every phase?
thats what i try to autodetect in the esp application, which phase is the most likely one.
(very crude auto detection)

Does your approach also work for three phase systems?

**Reply by gekkehenkie11** [reply]
> I wonder if the protocol you implemented is the same as the original one.

In the end the timing actually seems fine. So I just used stock firmware for the SAMD09, however, I modified it to take (V1-V2) * I instead of V * I. And it now finally produced the correct result for my devices that I connected between 2 phases.

> Didnt Emporia send the power calculation result against every phase?

correct. 

> Does your approach also work for three phase systems?

Well I don't have any 'approach' just yet ;) The 3 steps above are the way to do it. It's going to be some work though. I guess we first need to expand the YAML file with the option for users with 120 degrees V shift who connect (some of their) breakers between 2 phases. And then we need to implement that extra 'configuration' packet from ESP -> SAMD09, and then have the SAMD09 process that and use (V1-V2) for those situations. And for the situations where the user just connected the breaker between L+N, so the 'normal' situation, we need to do just what they now do already: V * I

**Comment by gekkehenkie11** [comment]
BTW so if the above is implemented, then it also becomes super easy to calculate power factor. Remember that the SAMD09 already sends the power values (for each phase) and the current RMS to the ESP. Once we have the correct real power, the apparent power is simply the RMS values of V * I , like the SAMD09 already sends to the ESP. So then we can simply divide real power by apparent power and there's your power factor.

Or in other words, for single phase, or 2 phase users (with 180 degrees shift of course) they can already do that. No need to mod the SAMD09 in that case. All that needs to be done in that case is have @flaviut's firmware multiply (V RMS) * (I RMS), like it already gets from the SAMD09 (!), and then divide the real power (which it also already gets from the SAMD09) by apparent power.

Actually, 3 phase users can also already do it now. But only as long as they connect breakers between L+N of course. But that's the 'normal' scenario anyway. Only some devices that need 220V-240V will be connected between 2 phases anyway. I think in my home it are actually only the air conditioners, and maybe my oven that need 220V. BTW, haha, that just made me think of this post: https://github.com/emporia-vue-local/esphome/discussions/2#discussioncomment-7296163 which now finally makes sense too :) His problem was not the power factor, but his connection between 2 phases with 120 degrees shift.

**Comment by gekkehenkie11** [comment]
Actually, thinking about it, the easiest way would be to just make the ESP packet bigger. And then instead of sending for each CT (what it does already now) ->

V1 * I
V2 * I
V3 * I

We simply add for the 16 smaller CT's:

(V1-V2) * I
(V2-V3) * I
(V1-V3) * I

So instead of sending 3 power value's per CT, we send 6. That way there's no need to send any 'configuration message' to the SAMD09. We can then simply determine in the ESP firmware which of the 6 above to use. That's of course the reason they're sending those 3 value's right now too. So we can just expand on that. The only question is if the SAMD09 is fast enough to do this, I already ran into speed problems, not sure if there's enough time left to add 48 times a subtraction and multiplication. Would have to find out about that. If not, we could then either lower the sample rate or bit, or implement the configuration packet like I suggested above.

Either way, changing the communication between ESP and SAMD09 would either mean we'd need @flaviut's firmware to be able to handle that, so there should be some YAML setting to let it know which of the 2 types of 'sensor reading' packets to expect.

**Comment by gekkehenkie11** [comment]
I'm using myself this now, it's quick n dirty but works really well: 

https://github.com/gekkehenkie11/Emporia-VUE-fix/tree/main

So I now have 8 ports configured for 2 phase connection and the others should be between phase and neutral. If you want to change this number you can simply change line 618 where it says 

if (Muxnr < 4)

to for example 

if (Muxnr < 2)

So then it would only use the first 4 ports for our special treatment. So if you only have 4 devices at home that need 2 phases then you could do it like that.

The port ordering is:

10, 2, 11, 3, 12, 4, 9, 1, 13, 5, 16, 8, 14, 6, 15, 7

So if you changed it into Muxnr < 2, it would then have ports 10, 2, 11 and 3 for 2 phases.

With some work it could become a lot nicer and easier (like described above) but for now I've invested already waaaay too much time into this and it works now very well. So for the moment I'll just leave it like this.

*EDIT* I've also just uploaded a precompiled version, ready to be flashed.

**Comment by gekkehenkie11** [comment]
LOL I actually have way more 220V devices in my house than I had thought :) All my split unit AC's, my solar, my oven, washing machine, coffee machine, tumble dryer, car charger and there are more that I haven't even identified are running between 2 phases. I now have one VUE completely flashed for 220V ports, so that also makes it easy to remember how it's all configured.

It's so damn awesome to FINALLY have correct value's. BTW I did find in all of my experiments that some of the CT's might have (slightly) higher or lower current measurements than others, so if after the above mod you still have some (slightly) different values than expected (like within 8%) you could try to change the CT, it fixed one last weird thing I saw.

**Comment by SamSavageSpeed** [comment]
Hello @gekkehenkie11,
Kudos to your brother & to you.
Would be great to have this somehow
integrated into ESPHome in Home Assistant.

**Reply by gekkehenkie11** [reply]
> Hello @gekkehenkie11, Kudos to your brother & to you. Would be great to have this somehow integrated into ESPHome in Home Assistant.

Thanks! Yeah it would be AWESOME if we could just flash it from HA. It should be possible, in fact @odewdney wrote a micro python tool to flash the chip via the ESP: https://github.com/odewdney/emporiavue

**Comment by SamSavageSpeed** [comment]
Hello @gekkehenkie11,

I have 2 devices in 2 different buildings which
are connected with 2 phases without neutral.

I just found "Phase B Phase Angle" for both :
Building one : 120° and Building two : 240°

About your workaround : (V1-V2)*I
Is it the same calculation for 240° ?

Thanks in advance.

**Reply by gekkehenkie11** [reply]
Hi yes that's not a problem at all. In fact I just added comments to make it a bit clearer when reading the code:
https://github.com/gekkehenkie11/Emporia-VUE-fix/commit/a036f32eff8b23170f1650e0f174bba62d3ba109

So if you select "phase_a" in the YAML file for the ct_clamps power, you will get the power based on the difference between phase a and phase b. If you select "phase_b" you will get the difference between phase a and phase c. And if you select "phase_c" ,the difference between phase b and phase c. So, for example, if your CT 9 is connected to a pole of a breaker that 's connected between phase a and phase c, you'd configure it like this in your YAML file:

{ phase_id: phase_b, input:  "9", power: { id:  cir9 } }

And if you're sure that's right, but the resulting power value is negative, you could just reverse the direction of the clamp, OR put it on the other pole OR use the 'abs' filter to always make the result positive.

So, how to know between which 2 phases your breaker is connected? I always start with first making 100% sure that it's between 2 phases (although the cable color SHOULD tell you, if correct LOL, I had one 3 pole breaker in my house that actually had 2 wires, neutral and phase! so 127V in my case) by measuring the voltage with a multimeter and make sure it's 220V-240V (if your single phase is on 110-130V). Then to find out between which phases it's connected you have 3 options:

1) You could trace the wires through your breaker panel. In my breaker panel that's next to impossible though. I have a 'European style' breaker panel and in that case it might be hard to trace the wires. 
2) Another option is to measure continuity with a multi meter, but you will need to power down your whole breaker panel at the source (you'll probably have a main switch but make sure it's all powered down or you'll blow things up). 
3) Or you could do it like I did, just try all 3: a b and c in the YAML file and simply see which one yields the correct power values. Of course you'll need to know what power to expect. I used this device myself: https://www.amazon.com/Amperage-Voltmeter-Multimeter-Transformer-Multi-Function/dp/B0BR75M7W1/ref=sr_1_3_sspa to verify. I just clamp it around a wire and then press the two voltage lines against the screws of the breakers to quickly test. And I also have a few of these buddies in my breaker panel installed: https://www.eastroneurope.com/products/view/sdm230dr which I also used, cause these are really accurate, since they're not using CT's but direct measurements. And for wall power outlets you can simply use something like this: https://www.amazon.com/Suraielec-90V-250V-Calculator-Protection-Electricity/dp/B0CR9Z19DQ . 

The reason that I don't like to power down everything is that I'm afraid to damage my fridge. These compressors can be really sensitive to flipping the power on/off so I try to avoid that.

**Reply by SamSavageSpeed** [reply]
Thanks for your reply @gekkehenkie11,

Would-it be possible to make such "shifted" Power calculation
directly in ESPHome yaml file in HA to keep the current firmware ?
https://gist.github.com/flaviut/93a1212c7b165c7674693a45ad52c512?permalink_comment_id=5080257#gistcomment-5080257

**Reply by gekkehenkie11** [reply]
Actually there is a super simple other way. I'll need to add this to the first post too :) My brother just told me "if you have 2 phases per breaker you could also simply use 2 CT's and then just add the power values up". LOL, I've tried this before with stock Emporia firmware but that never worked because the software would then choose the wrong phases. However with esphome we can just select the phases ourselves and I just checked it and it actually works :) The reason why this doesn't work with stock is that for example when my airconditioner is in standby, one of those 2 power values will get negative. So their algo would then ignore that one and choose the wrong one instead. But if you add up that negative to the positive value, that total indeed becomes a few watt in standby use so it actually works if you lock the phases correctly.

So the only disadvantage is that you need an extra CT. Which of course means actually the double amount of total CT's if you have 'fake' 2 phase like @flaviut has (120 degrees between them), so in those cases it makes definitely sense to flash the atmel firmware. But also in my own case, I have currently 4 VUE's at home and with this mod I have actually only 1 free port left, after wiring everything (big house, lots of breaker). So without flashing it, I would need another VUE. So the firmware mod saved me at least 1 VUE, haha.

**Reply by SamSavageSpeed** [reply]
Hello @gekkehenkie11,

Actually, on my 2nd Home Vue, I already did more or less what you describe to
monitor both legs of 4 breakers with 8 clamps but I was not sure if it was fine :

      # Pay close attention to set the phase_id for each breaker by matching it to the phase/leg it connects to in the panel
      - { phase_id: phase_b, input:  "1", power: { id:  cir1, filters: [ *pos, multiply: 1 ] } }
      - { phase_id: phase_a, input:  "2", power: { id:  cir2, filters: [ *pos, multiply: 1 ] } }
      - { phase_id: phase_b, input:  "3", power: { id:  cir3, filters: [ *pos, multiply: 1 ] } }
      - { phase_id: phase_a, input:  "4", power: { id:  cir4, filters: [ *pos, multiply: 1 ] } }
      - { phase_id: phase_b, input:  "5", power: { id:  cir5, filters: [ *pos, multiply: 1 ] } }
      - { phase_id: phase_a, input:  "6", power: { id:  cir6, filters: [ *pos, multiply: 1 ] } }
      - { phase_id: phase_b, input:  "7", power: { id:  cir7, filters: [ *pos, multiply: 1 ] } }
      - { phase_id: phase_a, input:  "8", power: { id:  cir8, filters: [ *pos, multiply: 1 ] } }
    on_update:
      then:
        - component.update: solar_power
        - component.update: cabinet_power
        - component.update: garage_prive_power
        - component.update: garage_location_power

      # The copy sensors filter and send the power state to HA        
      - { platform: copy, name: "Solar Power", source_id: solar_power, filters: *throttle_avg }
      - { platform: copy, name: "Cabinet Power", source_id: cabinet_power, filters: *throttle_avg }
      - { platform: copy, name: "Garage Privé Power", source_id: garage_prive_power, filters: *throttle_avg }
      - { platform: copy, name: "Garage Location Power", source_id: garage_location_power, filters: *throttle_avg }

  - platform: template
    name: "Solar Power"
    lambda: return id(cir1).state + id(cir2).state;
    update_interval: never   # will be updated after all power sensors update via on_update trigger
    id: solar_power
    device_class: power
    state_class: measurement
    unit_of_measurement: "W"
  - platform: total_daily_energy
    name: "Solar Daily Energy"
    power_id: solar_power
    accuracy_decimals: 0

  - platform: template
    name: "Cabinet Power"
    lambda: return id(cir3).state + id(cir4).state;
    update_interval: never   # will be updated after all power sensors update via on_update trigger
    id: cabinet_power
    device_class: power
    state_class: measurement
    unit_of_measurement: "W"
  - platform: total_daily_energy
    name: "Cabinet Daily Energy"
    power_id: cabinet_power
    accuracy_decimals: 0

  - platform: template
    name: "Garage Privé Power"
    lambda: return id(cir5).state + id(cir6).state;
    update_interval: never   # will be updated after all power sensors update via on_update trigger
    id: garage_prive_power
    device_class: power
    state_class: measurement
    unit_of_measurement: "W"
  - platform: total_daily_energy
    name: "Garage Privé Daily Energy"
    power_id: garage_prive_power
    accuracy_decimals: 0

  - platform: template
    name: "Garage Location Power"
    lambda: return id(cir7).state + id(cir8).state;
    update_interval: never   # will be updated after all power sensors update via on_update trigger
    id: garage_location_power
    device_class: power
    state_class: measurement
    unit_of_measurement: "W"
  - platform: total_daily_energy
    name: "Garage Location Daily Energy"
    power_id: garage_location_power
    accuracy_decimals: 0

That's good to know that it should be correct for this one :-) Here are last month's data to illustrate :
![image](https://github.com/emporia-vue-local/esphome/assets/154277260/bfcf3c0d-fef6-44ab-84be-219d88b26206)
NB : More than 4 circuits here because some of them are monitored by other measurement devices.

But, for my other Vue, I use the 16 clamps each on a single leg of breaker
so I should either monitor only 8 of my 16 actuals circuits or merge them.

- I already merged circuits because I have more than 16 breakers.
=> I will have to analyze what I want to monitor closely or not
and either merge more circuits or discard some circuits... :-/

**Reply by SamSavageSpeed** [reply]
Do you think Flaviut could/would integrate your findings in his project ?
To keep using a single clamp on a single leg to monitor more circuits :-)

**Reply by gekkehenkie11** [reply]
Well with a single CT you can't get correct value's if you have 120 degrees shift between the voltages, at least not without my firmware mod. I think the best way forward it so make it easy to flash the mod. @odewdney already wrote the script, it just needs integration into HA, someone would need to look into that. I don't have time for that though, but if that's available then it would become very easy to flash the mod and you could put a single CT on a 2 pole breaker with 2 phases.

And yes, your example YAML seems fine, you can add up the poles, just make sure the phases are correct.

**Reply by SamSavageSpeed** [reply]
Thanks for your answers/work/time. I will think about how to handle my first Vue with 16 clamps.
On one hand, I'm happy that you've confirmed that REAL power is actually mesured by Emporia
but, on the other hand, I'm very disappointed that they didn't adressed the shifting situation :-/

**Reply by gekkehenkie11** [reply]
Sure! BTW I found one mistake in your YAML, you should not use "POS" filter when adding up 2 phases. Then you'll make the same mistake as Emporia does in their firmware. Don't use pos for those CT's, make sure your have the right phases and that your CT's are correctly oriented. It's very well possible that one CT might become negative, especially when an air conditioner is in standby for example.

**Comment by SamSavageSpeed** [comment]
Thanks for your feedback. So I should remove *pos in the filters here below ?

      - { phase_id: phase_b, input:  "1", power: { id:  cir1, filters: [ *pos, multiply: 1 ] } }
      - { phase_id: phase_a, input:  "2", power: { id:  cir2, filters: [ *pos, multiply: 1 ] } }
      - { phase_id: phase_b, input:  "3", power: { id:  cir3, filters: [ *pos, multiply: 1 ] } }
      - { phase_id: phase_a, input:  "4", power: { id:  cir4, filters: [ *pos, multiply: 1 ] } }
      - { phase_id: phase_b, input:  "5", power: { id:  cir5, filters: [ *pos, multiply: 1 ] } }
      - { phase_id: phase_a, input:  "6", power: { id:  cir6, filters: [ *pos, multiply: 1 ] } }
      - { phase_id: phase_b, input:  "7", power: { id:  cir7, filters: [ *pos, multiply: 1 ] } }
      - { phase_id: phase_a, input:  "8", power: { id:  cir8, filters: [ *pos, multiply: 1 ] } }
      
Maybe even remove completly filters since multiply by 1 doesn't do anything ?

      - { phase_id: phase_b, input:  "1", power: { id:  cir1 } }
      - { phase_id: phase_a, input:  "2", power: { id:  cir2 } }
      - { phase_id: phase_b, input:  "3", power: { id:  cir3 } }
      - { phase_id: phase_a, input:  "4", power: { id:  cir4 } }
      - { phase_id: phase_b, input:  "5", power: { id:  cir5 } }
      - { phase_id: phase_a, input:  "6", power: { id:  cir6 } }
      - { phase_id: phase_b, input:  "7", power: { id:  cir7 } }
      - { phase_id: phase_a, input:  "8", power: { id:  cir8 } }

**Reply by gekkehenkie11** [reply]
Correct. And then make sure it's all configured correctly, meaning that you need to be sure that the phases are correct and that the values make sense. For example I have an AC between 2 phases, I get for example +30 on the one phase and -22 on the other in standby, resulting 8 Watt, which I know is correct. And when I turn the AC on, both get positive which is of course correct too. As a rule of thumb, it seems to be true that if you connect between 2 phases, one of the CT's need to be pointing in the other direction as the other one. So one arrow K->L should point toward the breaker and the other towards the load.

BTW in general I would advise not to use filters at all, unless you're 100% sure that what you're doing is correct. I find it way easier to spot mistakes when there are no filters. I do use some 'abs' myself for example, in those cases that I'm absolutely 100% sure that the CT is installed in the wrong direction, on a L+N breaker (!) and when it's difficult to take them out and switch direction, because there's no space for example. But again, when connecting between 2 phases you should never do that since a negative value on one of the 2 poles actually might be the correct outcome.

**Reply by SamSavageSpeed** [reply]
At the moment, the 8 clamps are put in the same direction on all legs of the 4 breakers.
Maybe I could keep the multiply filter afterall and just multiply by "-1" one of the leg ?
This way I could avoid changing orientation of the clamps ? Giving something like this:

  - { phase_id: phase_b, input:  "1", power: { id:  cir1, filters: [ multiply: 1 ] } }
  - { phase_id: phase_a, input:  "2", power: { id:  cir2, filters: [ multiply: -1 ] } }
  - { phase_id: phase_b, input:  "3", power: { id:  cir3, filters: [ multiply: 1 ] } }
  - { phase_id: phase_a, input:  "4", power: { id:  cir4, filters: [ multiply: -1 ] } }
  - { phase_id: phase_b, input:  "5", power: { id:  cir5, filters: [ multiply: 1 ] } }
  - { phase_id: phase_a, input:  "6", power: { id:  cir6, filters: [ multiply: -1 ] } }
  - { phase_id: phase_b, input:  "7", power: { id:  cir7, filters: [ multiply: 1 ] } }
  - { phase_id: phase_a, input:  "8", power: { id:  cir8, filters: [ multiply: -1 ] } }
  
  Off course, on each of the 4 breakers, I should try some expected loads to verify.

**Reply by gekkehenkie11** [reply]
yeah multiplying with -1 is indeed an option.

**Comment by BlipRanger** [comment]
Following this thread. I just recently finished flashing my Gen 3 (NA with a handful of 220v breakers, only measuring a single leg) with this firmware and noticed that when compared to what was being reported by my power company, it was off by quite a bit. 
So 2 quick questions: 
- Does this same issue affect gen 3?
- Does the measure of power (as used here) differ much from consumption measured upstream by a power company?

**Reply by gekkehenkie11** [reply]
Yeah same thing with Gen 3. I have a Gen3 here to test. Haven't dumped the Atmel firmware yet but I suspect it's the same code.

the consumption measured by the power company is quite similar as what the Emporia indicates. It's actually quite accurate. But keep in mind that that's being measured by the 3 big main CT's, these were ok to begin with. This thread is specifically about the 16 smaller CT's. Or better said, situations where you use 2 (or  even 3) phases on a single breaker AND when there's 120 degrees difference between the voltages.

**Reply by BlipRanger** [reply]
Oh, OK I understand now. I must be doing something wrong then because sometimes I see as much as a 3X (lower) difference between the big CT and what's being reported by the power company.

**Reply by AndreKR** [reply]
What does "use 2 (or even 3) phases on a single breaker" mean? You can't exactly connect two phases to one breaker without starting a fire. Do you mean the problem appears when a single device is actually using 3-phase power, like a big motor or something? And adding up the power values from the three phases wouldn't work?

**Reply by gekkehenkie11** [reply]
Oh you can. It's actually a common way on a 2 phase or 3 phase system to generate a different voltage without needing a transformer. For example in my house I'm getting 3 phases with 127V between phase and neutral from my utility company. And in my case it's true 3 phases, so I'm getting 120 degrees between the phases. If I supply two phases (different ones) to one breaker the resulting voltage is 127V * V3 = 220V. This is a common way in my country to for example power an air conditioner, which usually is 220V-ish. Also see the chart in my first post which shows the resulting voltage.

In the USA this is a common way to generate 220V when there are 2 phases (180 degrees) and there's 110V between neutral and phase. However in this situation this thread is not applicable. This thread is about the situations when there's a 120 degrees between the phases. Why? Think of the chart in the first post. If there's 180 degrees between them, then the resulting chart actually is like a multiplication of either charts by factor 2. So you can just use factor 2 in emporia in those cases and then just measure one leg. But again, you can't do that with a 120 degrees shift, that's what this is all about.

**Reply by AndreKR** [reply]
What? Really? Like this?
![image](https://github.com/user-attachments/assets/06507c05-fee6-4da6-ab5b-c441b1ac6940)

**Reply by flaviut** [reply]
[North American residential power is typically split-phase.](https://en.wikipedia.org/wiki/Split-phase_electric_power) But I think that's unrelated to the point.

This style of breaker (the 50A), where two phases are mechanically linked in the same breaker is extremely common in NA:

![Insert-the-hot-wire-700x525](https://github.com/user-attachments/assets/9e6a5b72-d52d-4f5a-9ca8-5e6532290f2c)

If either phase trips, the whole breaker trips. The overall voltage between the two wires (note they are black & red, for two hots, rather than black & white for hot+neutral) is 240V or 208V depending on whether the installation is split-phase or 2-of-3-phases.

**Reply by AndreKR** [reply]
Ok, so what I would call "two linked breakers" is actually called "one breaker", but it's still two separate switches and the phases are never shorted to each other.

Unfortunately that didn't really help me to understand in which situations the Vue works and in which situations it doesn't work (and *how* it fails in those cases).

**Reply by gekkehenkie11** [reply]
No don't short it, that WILL result in a fire indeed. It's indeed like @flaviut just explained.

Like explained, the VUE won't work in situations where you're getting multiple phases from your utility company and you have 120 degrees between 2 phases and feed those 2 phases to a device. So instead of feeding L+N, you feed L1+L2. Well, actually it WILL work, as long as you use esp home, put a CT on each leg, and lock the right phases to each leg and then sum the 2 legs (or use this mod as explained here and you can only monitor one leg, saving a CT). But all this won't work with stock firmware since the emporia is very likely to select the wrong phase(s) in this situation. Before understanding all this, I used to use Emporia stock firmware and my 220V air conditioner would for example show a 90 watt consumption in standby which is completely wrong. There is a long thread about this on the emporia forums. The first theory was that the emporia measured apparent power instead of real power. But that turned out to be wrong and the real issue is described in this thread.

Anyway I'd highly suggest not to experiment with all this stuff if you don't completely understand what you're doing.

**Reply by AndreKR** [reply]
So, to be clear, if I use ESPHome I don't need to flash the Atmel chip with the alternative firmware?

**Reply by flaviut** [reply]
~If you are not an experienced embedded software engineer looking to tinker, do not try and flash the Atmel chip.~

~If you flash the Atmel, I'm not even sure it's possible to go back to the original…~

See below

**Reply by gekkehenkie11** [reply]
@AndreKR You don't need to flash the Atmel chip. The advantage though is that by flashing it you can save some ports since you only need 1 CT per leg.
@flaviut Nah it's really not that much more difficult than flashing the ESP chip. The only thing is of course that you need some jlink connector. During the coding of my alternative firmware I messed up tons of times, resulting in a bricked device, but every time it was super easy to flash a new version and unbrick it. I dont think you can do much harm.

**Reply by flaviut** [reply]
I'm sorry about the FUD! I still don't think this is a beginner thing, but I was worried about being unable to backup the stock firmware image. Thanks for cleaning things up

**Comment by mmmgggttt** [comment]
hello guys @gekkehenkie11 , i'm trying to debugging my installation , where i was initially facing issue with real/apparent power with Emporia stock fw (whenever an inverter was installed), which i though i got read with Esp32 local firmware.

My grid inlet is 400v 3phases 120° and 230v from each phase to neutral (central star). I then have some 3phases appliances (heatpump and elevator motor ) and other single phase which are then balances across the 3 phases  L1-N , L2-N and L3-N.

Do you believe in such setup i can run the standard config or not? At the moment seems i'm getting good readings from my CT but im not a 100% sure....
